src/common/misc.c

```diff
@@ -249,6 +249,8 @@ daos_sgl_get_bytes(d_sg_list_t *sgl, struct daos_sgl_idx *idx,
 			size_t buf_len_req,
 			uint8_t **buf, size_t *buf_len)
 {
+	D_ASSERT(idx->iov_idx < sgl->sg_nr);
```

Comment:
do we validate these elsewhere?

Reply:
No, and I don't remember why I added that. Probably to catch a programmer error in something I was doing. I'm going to change it so it just returns if that's the case.